airbyte-ci/connectors/base_images/base_images/hacks.py

```diff
@@ -0,0 +1,27 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+import dagger
+
+
+def get_container_dockerfile(container) -> str:
+    """Returns the Dockerfile of the base image container.
+    Disclaimer: THIS IS HIGHLY EXPERIMENTAL, HACKY AND BRITTLE.
+    TODO: CONFIRM WITH THE DAGGER TEAM WHAT CAN GO WRONG HERE.
+    Returns:
+        str: The Dockerfile of the base image container.
+    """
+    lineage = [
+        field for field in list(container._ctx.selections) if isinstance(field, dagger.api.base.Field) and field.type_name == "Container"
+    ]
+    dockerfile = []
+    for field in lineage:
+        if field.name == "from":
```

Comment:
💡 Were parsing a DSL here!

A good pattern for that **can** include a dictionary of token -> function

```py
daggerTokenToDocker = {
    "withExec": lambda field : f'RUN {" ".join(field.args.get("args"))}',
}

for field in lineage:
    dockerfile.append(daggerTokenToDocker[field.name](field))
```

Reply:
Yep I found it could indeed leverage a mapping. Will do.